spotify_db/Query.sql

```diff
@@ -0,0 +1,169 @@
+-- Top popular tracks
+SELECT t.track_name, ar.artist_name, al.album_name, t.popularity
+FROM tracks t
+JOIN artists ar ON t.artist_id = ar.artist_id
+JOIN albums al ON t.album_id = al.album_id
+WHERE t.popularity > (SELECT AVG(popularity) FROM tracks)
+ORDER BY t.popularity DESC; 
+
+-- Artists 
+SELECT 
+    ar.artist_name,
+    ar.followers,
+    COUNT(DISTINCT t.track_id) AS track_count,
+    COUNT(DISTINCT t.album_id) AS album_count,
+    COUNT(DISTINCT pt.playlist_id) AS playlist_count,
+    ar.popularity AS artist_popularity,
+    AVG(t.popularity) AS avg_track_popularity,
+    AVG(al.popularity) AS avg_album_popularity
```

Comment:
Let's order all dimensions together, then aggregations.

```suggestion
    ar.artist_name,
    ar.followers AS artist_followers,
    ar.popularity AS artist_popularity,
    COUNT(DISTINCT t.track_id) AS track_count,
    COUNT(DISTINCT t.album_id) AS album_count,
    COUNT(DISTINCT pt.playlist_id) AS playlist_count,
    AVG(t.popularity) AS avg_track_popularity,
    AVG(al.popularity) AS avg_album_popularity
```